lib/number-to-words.hoon

```diff
@@ -0,0 +1,105 @@
+=/  one-decillion          1.000.000.000.000.000.000.000.000.000.000.000
+=/  one-undecillion        1.000.000.000.000.000.000.000.000.000.000.000.000
+=/  one-duodecillion       1.000.000.000.000.000.000.000.000.000.000.000.000.000
+=/  one-tredecillion       1.000.000.000.000.000.000.000.000.000.000.000.000.000.000
+=/  one-quattuordecillion  1.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000
+=/  one-quindecillion      1.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000
+=/  one-sexdecillion       1.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000
+=/  one-septendecillion    1.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000
+=/  one-octodecillion      1.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000
+=/  one-novemdecillion     1.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000
+=/  one-vigintillion       1.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000
+=/  max                    1.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000.000
+|%
+::  +eng:number-to-words: convert @u to english
+::
+++  eng
+  |=  num=@u
+  ^-  (unit tape)
+  ::  0-19
+  ?:  =(num 0)   [~ "zero"]
+  ?:  =(num 1)   [~ "one"]
+  ?:  =(num 2)   [~ "two"]
+  ?:  =(num 3)   [~ "three"]
+  ?:  =(num 4)   [~ "four"]
+  ?:  =(num 5)   [~ "five"]
+  ?:  =(num 6)   [~ "six"]
+  ?:  =(num 7)   [~ "seven"]
+  ?:  =(num 8)   [~ "eight"]
+  ?:  =(num 9)   [~ "nine"]
+  ?:  =(num 10)  [~ "ten"]
+  ?:  =(num 11)  [~ "eleven"]
+  ?:  =(num 12)  [~ "twelve"]
+  ?:  =(num 13)  [~ "thirteen"]
+  ?:  =(num 14)  [~ "fourteen"]
+  ?:  =(num 15)  [~ "fifteen"]
+  ?:  =(num 16)  [~ "sixteen"]
+  ?:  =(num 17)  [~ "seventeen"]
+  ?:  =(num 18)  [~ "eighteen"]
+  ?:  =(num 19)  [~ "nineteen"]
+  ::  20-99
+  =/  tpl  (div num ten)                                ::  tens place
+  =/  rem  (mod num ten)                                ::  ones place
+  =/  sfx                                               ::  suffix
+    ?:  |(=(rem 0) (gte tpl 10))
+      ~
+    ['-' (need $(num rem))]
+  ?:  =(tpl 2)  [~ (weld "twenty" sfx)]
+  ?:  =(tpl 3)  [~ (weld "thirty" sfx)]
+  ?:  =(tpl 4)  [~ (weld "fourty" sfx)]
+  ?:  =(tpl 5)  [~ (weld "fifty" sfx)]
+  ?:  =(tpl 6)  [~ (weld "sixty" sfx)]
+  ?:  =(tpl 7)  [~ (weld "seventy" sfx)]
+  ?:  =(tpl 8)  [~ (weld "eighty" sfx)]
+  ?:  =(tpl 9)  [~ (weld "ninety" sfx)]
+  ::  100-max
+  =/  num-break                                         ::  repeated pattern
+    ::  min: minimum to qualify for this break
+    ::  str: english word for this break
+    ::  sep: separator between this and the previous break
+    |=  [min=@u str=tape sep=tape]
+    =/  rem  (mod num min)
+    :-  ~
+    %+  weld
+      (weld (need ^$(num (div num min))) [' ' str])
+    ?:(=(rem 0) ~ (weld sep (need ^$(num rem))))
```

Comment:
This line feels a bit crammed. Tall-forming the `?:` would probably help readability.